service/role/service.go

```diff
@@ -0,0 +1,114 @@
+package role
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	vaultclient "github.com/hashicorp/vault/api"
+)
+
+// Config defines configurable aspects (such as dependencies) of this service.
+type Config struct {
+	// Config
+	PKIMountpoint string
+
+	// Dependencies.
+	VaultClient *vaultclient.Client
+}
+
+// DefaultConfig returns a default configuration that can be used to create this service.
+func DefaultConfig() Config {
+	vaultClientConfig := vaultclient.DefaultConfig()
+	vaultClientConfig.Address = "http://127.0.0.1:8200"
+	vaultClient, err := vaultclient.NewClient(vaultClientConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	config := Config{
+		// Dependencies.
+		VaultClient: vaultClient,
+	}
+
+	return config
+}
+
+// New takes a configuration and returns a configured service.
+func New(config Config) (Service, error) {
+	// Dependencies.
+	if config.VaultClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "Vault client must not be empty")
+	}
+
+	if config.PKIMountpoint == "" {
+		return nil, microerror.Maskf(invalidConfigError, "PKIMountpoint must not be empty")
+	}
+
+	service := &service{
+		Config: config,
+	}
+
+	return service, nil
+}
+
+type service struct {
+	Config
```

Comment:
This is the config. 

```
type Config struct {
	// Dependencies.
	VaultClient *vaultclient.Client

	// Settings.
	PKIMountpoint string
}
```

Then this is the service. 

```
type Service struct {
	// Dependencies.
	vaultClient *vaultclient.Client

	// Settings.
	pkiMountpoint string
}
```

We do this in a lot of places already. It is just that the certctl code is already quite old and back then I did not have the conclusion I have now.